tests/components/enocean/test_light.py

```diff
@@ -0,0 +1,38 @@
+"""Test EnOcean light implementation."""
+import unittest
+from unittest.mock import MagicMock, patch
+
+from enocean.protocol.packet import Packet
+
+from homeassistant.components import enocean as enocean_component
+from homeassistant.components.enocean import light
+
+
+class TestEnOceanLight(unittest.TestCase):
+    """Verify that the light can understand EnOcean packets."""
+
+    def setUp(self):
+        """Set up things to be run when tests are started."""
+        enocean_component.ENOCEAN_DONGLE = MagicMock()
+        self.sender_id = [0xAA, 0xAA, 0xAA, 0xAA]
+        self.dev_id = [0xBB, 0xBB, 0xBB, 0xBB]
+        self.device = light.EnOceanLight(
+            self.sender_id, self.dev_id, 'light name')
+
+    def test_setup(self):
+        """Test initialization."""
+        assert self.device.name == 'light name'
```

Comment:
Don't assert anything about the entity directly but use the core state machine to get the state of the entity and assert the state. The state is the truth of the entity.

Avoiding direct entity interaction in tests will make the tests more robust.

Reply:
Tests removed